src/DynamoUtilities/PathHelper.cs

```diff
@@ -146,5 +147,17 @@ public static bool isValidJson(string path, out string fileContents, out Excepti
                 return false;
             }
         }
+
+        // Check if the file name contains any special non-printable chatacters.
+        public static bool IsFileNameInValid(string fileName)
+        {
+            // Some other extra characters that are to be checked. 
+            char[] invalidCharactersFileName = { '#', '%', '&', '.', ' ' };
+
+            if (fileName.Any(f => Path.GetInvalidFileNameChars().Contains(f)) || fileName.IndexOfAny(invalidCharactersFileName) > -1)
```

Comment:
@reddyashish Can you add a comment linking to MSDN or something similar showing which characters are checked in `GetInvalidFileNameChars` vs the ones you have manually added?  I think you may have posted it in an earlier comment but don't see it anymore.

Reply:
This Microsoft doc just shows some of the characters that are not valid for a file name. 
https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getinvalidpathchars?view=netframework-4.7.2

I am also attaching a screenshot of all the characters in the array returned by GetInvalidFileNameChars() function. 

<img width="1187" alt="screen shot 2018-12-07 at 1 28 08 pm" src="https://user-images.githubusercontent.com/43763136/49635100-afccc780-fa24-11e8-8200-db7f96ab91f4.png">

The extra characters that were checked manually are { '#', '%', '&', '.', ' ' }.